CMakeLists.txt

```diff
@@ -855,29 +761,156 @@ set(devilutionx_assets
   ui_art/mainmenuw.pcx
   ui_art/supportw.pcx)
 
+if(VIRTUAL_GAMEPAD)
+  list(APPEND devilutionx_assets
+    ui_art/button.png
+    ui_art/directions2.png
+    ui_art/directions.png
+    ui_art/menu-levelup.png
+    ui_art/menu.png)
+endif()
+
+if(NOT DEFINED DEVILUTIONX_ASSETS_OUTPUT_DIRECTORY)
+  set(DEVILUTIONX_ASSETS_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/assets")
+endif()
 
 # Copy assets to the build assets subdirectory. This serves two purposes:
 # - If smpq is installed, devilutionx.mpq is built from these files.
 # - If smpq is not installed, the game will load the assets directly from this directoy.
 foreach(asset_file ${devilutionx_assets})
   set(src "${CMAKE_CURRENT_SOURCE_DIR}/Packaging/resources/assets/${asset_file}")
   set(dst "${DEVILUTIONX_ASSETS_OUTPUT_DIRECTORY}/${asset_file}")
-  list(APPEND DEVILUTIONX_MPQ_FILES "${asset_file}")
   list(APPEND DEVILUTIONX_OUTPUT_ASSETS_FILES "${dst}")
+  list(APPEND DEVILUTIONX_MPQ_FILES "${asset_file}")
   add_custom_command(
     COMMENT "Copying ${asset_file}"
     OUTPUT "${dst}"
     DEPENDS "${src}"
     COMMAND ${CMAKE_COMMAND} -E copy "${src}" "${dst}"
     VERBATIM)
+
+  if(APPLE)
+    list(APPEND BIN_TARGET_SRCS ${dst})
+    string(REGEX MATCH "(.*)\/" subpath ${asset_file})
+    set_source_files_properties(${dst} PROPERTIES
+      MACOSX_PACKAGE_LOCATION "Resources/${subpath}"
+      XCODE_EXPLICIT_FILE_TYPE compiled)

```

Comment:
On Mac, might as well not copy the file at all but set this property on `${src}` instead (without XCODE_EXPLICIT_FILE_TYPE compiled).

Reply:
If we do not set XCODE_EXPLICIT_FILE_TYPE selecting the file in Xcode wil try to load it as text which can freeze Xcode.